src/modules/roc_sndio/target_sox/roc_sndio/sox_backend.cpp

```diff
@@ -196,5 +196,22 @@ ISource* SoxBackend::open_source(core::IAllocator& allocator,
     return source.release();
 }
 
+void SoxBackend::get_drivers(core::Array<DriverInfo>& arr, ProbeFlags driver_type) {
+    const sox_format_tab_t* formats = sox_get_format_fns();
+    char const* const* format_names;
+    for (size_t n = 0; formats[n].fn; n++) {
+        sox_format_handler_t const* handler = formats[n].fn();
+        if ((!(handler->flags & SOX_FILE_DEVICE)) & (size_t)(ProbeFile & driver_type) >> 2
+            || ((handler->flags & SOX_FILE_DEVICE) && !(handler->flags & SOX_FILE_PHONY))
+                & (size_t)(ProbeDevice & driver_type) >> 3) {
```

Comment:
If I understand correctly, this:

```
(size_t)(ProbeFile & driver_type) >> 2
```

and this:

```
(size_t)(ProbeDevice & driver_type) >> 3
```

both are equal to `1`, which is a bit strange code. Also, it will silently break if the enum definition will change.

We can rewrite it like this:

```
if (((handler->flags & SOX_FILE_DEVICE) != 0) == ((ProbeFile & driver_type) != 0)) || ...) {
   ...
}
```

Arguably, we can also make it more readable by extracting a variable, like:

```
bool match;
if (driver_type & ProbeFile) {
    match = (handler->flags & SOX_FILE_DEVICE);
} else if (driver_type & ProbeDevice) {
    match = (handler->flags & SOX_FILE_DEVICE) && !(handler->flags & SOX_FILE_PHONY);
}
if (match) {
   ...
}
```

(or we can extract a function `bool match(handle)`)

Also, I suggest to rename IBackend::ProbeDevice and other flags to "FilterDevice" or "MatchDevice" or something like this. Previously these flags were used only in probe(), but now they're also used here.